langchain/vectorstores/matching_engine.py

```diff
@@ -80,19 +87,24 @@ def __init__(
         self.index = index
         self.endpoint = endpoint
         self.embedding = embedding
-        self.gcs_client = gcs_client
         self.credentials = credentials
+        self.gcs_client = gcs_client
         self.gcs_bucket_name = gcs_bucket_name
+        self.firestore_client = firestore_client
+        self.firestore_collection_name = firestore_collection_name
+        if not (firestore_collection_name or gcs_bucket_name):
+            raise ValueError("Storage not specified. Either a firestore collection or gcs bucket must be provided!")
 
     def _validate_google_libraries_installation(self) -> None:
         """Validates that Google libraries that are needed are installed."""
         try:
             from google.cloud import aiplatform, storage  # noqa: F401
             from google.oauth2 import service_account  # noqa: F401
+            from google.cloud import firestore
```

Comment:
if someone was using gas and didn't have firestore installed, this would raise an error right?

Reply:
@baskaryan Good catch! fixed now.